test/games/strategy/engine/data/UnitCollectionTest.java

```diff
@@ -0,0 +1,386 @@
+    assertThat(uc.getUnitCount(), is(equalTo(expUnitCount)));
+  }
+
+  @Test
+  public void getUnitCountByUnitType() {
+    assertThat(uc.getUnitCount(unitTypeOne), is(equalTo(0)));
+    assertThat(uc.getUnitCount(unitTypeTwo), is(equalTo(0)));
+
+    addAllDefaultPlayerUnits();
+    assertThat(uc.getUnitCount(unitTypeOne), is(equalTo(unitCountDefaultPlayerUnitTypeOne)));
+    assertThat(uc.getUnitCount(unitTypeTwo), is(equalTo(unitCountDefaultPlayerUnitTypeTwo)));
+
+    addAllOtherPlayerUnits();
+    assertThat(uc.getUnitCount(unitTypeOne), is(equalTo(unitCountUnitTypeOne)));
+    assertThat(uc.getUnitCount(unitTypeTwo), is(equalTo(unitCountUnitTypeTwo)));
+  }
+
+  @Test
+  public void getUnitCountByUnitTypeAndPlayerID() {
+    assertThat(uc.getUnitCount(unitTypeOne, defaultPlayerID), is(equalTo(0)));
+    assertThat(uc.getUnitCount(unitTypeTwo, defaultPlayerID), is(equalTo(0)));
+    assertThat(uc.getUnitCount(unitTypeOne, otherPlayerID), is(equalTo(0)));
+    assertThat(uc.getUnitCount(unitTypeTwo, otherPlayerID), is(equalTo(0)));
+
+    addAllDefaultPlayerUnits();
+    assertThat(uc.getUnitCount(unitTypeOne, defaultPlayerID), is(equalTo(unitCountDefaultPlayerUnitTypeOne)));
+    assertThat(uc.getUnitCount(unitTypeTwo, defaultPlayerID), is(equalTo(unitCountDefaultPlayerUnitTypeTwo)));
+    assertThat(uc.getUnitCount(unitTypeOne, otherPlayerID), is(equalTo(0)));
+    assertThat(uc.getUnitCount(unitTypeTwo, otherPlayerID), is(equalTo(0)));
+
+    addAllOtherPlayerUnits();
+    assertThat(uc.getUnitCount(unitTypeOne, defaultPlayerID), is(equalTo(unitCountDefaultPlayerUnitTypeOne)));
+    assertThat(uc.getUnitCount(unitTypeTwo, defaultPlayerID), is(equalTo(unitCountDefaultPlayerUnitTypeTwo)));
+    assertThat(uc.getUnitCount(unitTypeOne, otherPlayerID), is(equalTo(unitCountOtherPlayerUnitTypeOne)));
+    assertThat(uc.getUnitCount(unitTypeTwo, otherPlayerID), is(equalTo(unitCountOtherPlayerUnitTypeTwo)));
+  }
+
+  @Test
+  public void getUnitCountByPlayerID() {
+    assertThat(uc.getUnitCount(defaultPlayerID), is(equalTo(0)));
+    assertThat(uc.getUnitCount(otherPlayerID), is(equalTo(0)));
+
+    addAllDefaultPlayerUnits();
+    assertThat(uc.getUnitCount(defaultPlayerID), is(equalTo(unitCountDefaultPlayer)));
+    assertThat(uc.getUnitCount(otherPlayerID), is(equalTo(0)));
+
+    addAllOtherPlayerUnits();
+    assertThat(uc.getUnitCount(defaultPlayerID), is(equalTo(unitCountDefaultPlayer)));
+    assertThat(uc.getUnitCount(otherPlayerID), is(equalTo(unitCountOtherPlayer)));
+  }
+
+  @Test
+  public void containsAll() {
+    final Collection<Unit> unitsOfDefaultPlayerOfUnitTypeOne = getUnitsOfDefaultPlayerOfUnitTypeOne();
+    assertThat(uc.containsAll(unitsOfDefaultPlayerOfUnitTypeOne), is(equalTo(false)));
+    addAllDefaultPlayerUnits();
+    assertThat(uc.containsAll(unitsOfDefaultPlayerOfUnitTypeOne), is(equalTo(true)));
+  }
+
+  @Test
+  public void getUnitsByUnitTypeAndMaxValue() {
+    addAllDefaultPlayerUnits();

```

Comment:
Code smell here, programming by side effects.
Better to return a set:
```
Type unitCollection = addAllDefaultPlayerUnits();
assertThat(unitCollection.getUnits...)
```
This way a person neither needs to remember which variable is updated by addAllDefaultPlayerUnits, nor do you have to do a search to find it either.